graph/src/blockchain/firehose_block_stream.rs

```diff
@@ -101,11 +101,32 @@ fn stream_blocks<C: Blockchain, F: FirehoseMapper<C>>(
     // block of the longuest chain creating inconsistencies in the data (because it would
     // not revert the forked the block).
     //
-    // We should perform that only if subgraph actually started from a subgraph block ptr
-    // and no Firehose cursor was present. If a Firehose cursor is present, it's used to
-    // resume and as such, there is no need to perform this check (at the same time, it's
-    // not a bad check to make).
-    let mut check_subgraph_continuity = latest_cursor == "" && subgraph_current_block.is_some();
+    // If a Firehose cursor is present, it's used to resume the stream and as such, there is no need to
+    // perform the chain continuity check.
+    //
+    // If there was no cursor, now we need to check if the subgraph current block is set to something.
+    // When the graph node deploys a new subgraph, it always create a subgraph ptr for this subgraph, the
+    // initial subgraph block pointer points to the parent block of the manifest's start block, which is usually
+    // equivalent (but not always) to manifest's start block number - 1.
+    //
+    // Hence, we only need to check the chain continuiy if the subgraph block ptr is higher or equal to the
+    // subgraph actual block pointer, indeed, only in this case (and there is no firehose cursor) it means the
```

Comment:
> subgraph actual block pointer

Is this referring to the start block number?

Reply:
Yeah it's a mistake in the comment you are right, it should be `manifest's start block number` here